src/libsync/networkjobs.cpp

```diff
@@ -911,4 +911,36 @@ bool SimpleNetworkJob::finished()
     return true;
 }
 
+void fetchPrivateLinkUrl(AccountPtr account, const QString &remotePath,
+    const QByteArray &numericFileId, QObject *target,
+    std::function<void(const QString &url)> targetFun)
+{
+    const QString oldUrl = account->deprecatedPrivateLinkUrl(numericFileId).toString(QUrl::FullyEncoded);
+
+    // If the server doesn't have the property, use the old url directly.
+    if (!account->capabilities().privateLinkPropertyAvailable()) {
+        QTimer::singleShot(0, target, [=]() {
```

Comment:
@ogoffart Maybe Qt should have an overload for https://doc.qt.io/qt-5/qmetaobject.html taking a lambda with QueuedConnection?

Reply:
It has in Qt 5.10